subsys/bluetooth/controller/Kconfig.ll_sw_split

```diff
@@ -211,6 +211,13 @@ config BT_CTLR_SCAN_SYNC_ISO_SET
 	help
 	  Maximum supported broadcast isochronous groups (BIGs) sync sets.
 
+config BT_CTLR_ADV_ENABLE_STRICT
+	bool "Enforce Strict Advertising Enable/Disable"
```

Comment:
Why add option that allows to break qualification?

Reply:
This option had been the old behavior and does not break BT v4.2. The relaxed LE advertising enable was introduced in V5.0. Zephyr still keep the advertising state, and this option can be used by combined host+controller applications to valid their state machines.

Reply:
I agree with @sjanc here. 

If we can disable this, and thus not be BT 5 compliant, then we should at least set the reported BT version to that version, and alot disallow any BT5 and newer features. 

If we don't want to do that, I don't think we should allow this Kconfig option. 

There was another similar instance, where we actually, instead of removing it, added a warning if it was enabled, as it would break compliance and should not be used in a production build.

Reply:
Just as the implementation can be changed by users to make the stack non-compliant, this Kconfig should not be seen any different. This Kconfig cannot be turned on by simply having it in the project conf file, it is only visible if advanced features of the controller is enabled explicitly. Even if enabled or disabled does not fail any latest HCI conformance test cases.

Reply:
@cvinayak could you please state what the purpose of this change is? Is it to allow users to qualify the controller under the 4.2 spec?

Reply:
1. Qualification under v4.2 spec is valid under latest TCRL 2021-1, simple application like beacon can use it still, disabling privacy and filters reduce code size by 5KB.
2. Advertising enable has one parameter and we do not change (generate) private addresses on advertising enable, not necessary for different return codes, see @joerchan comment: https://github.com/zephyrproject-rtos/zephyr/pull/36446#discussion_r656155749 and https://github.com/zephyrproject-rtos/zephyr/pull/36446#discussion_r656220205
3. Zephyr Host does not call advertising enable if it is already enabled, this option compliments this if required (say to debug)